provisioning_templates/iPXE/kickstart_default_ipxe.erb

```diff
@@ -16,21 +16,36 @@ oses:
   <% static = (@host.token.nil? ? '?' : '&') + static_arg -%>
 <% end -%>
 
+<%
+options = []
+hostname = @host.name
+iface = @host.provision_interface
+mac = @host.provision_interface.mac
+-%>
+
 <% stage2 = host_param('kickstart_liveimg') ? 'inst.stage2=' + @host.operatingsystem.medium_uri(@host).to_s : '' %>
 
-<%- if (@host.operatingsystem.name.match(/Fedora/i) && @host.operatingsystem.major.to_i < 17) || @host.operatingsystem.major.to_i < 7 %>
-<%- net_options = 'ip=${netX/ip} netmask=${netX/netmask} gateway=${netX/gateway} dns=${dns}' -%>
-<%- else -%>
-<%- net_options = 'ip=${netX/ip}::${netX/gateway}:${netX/netmask}:::none nameserver=${dns}' -%>
-<%- end -%>
+<%- if (@host.operatingsystem.name.match(/Fedora/i) && @host.operatingsystem.major.to_i < 17) || @host.operatingsystem.major.to_i < 7
+      options.push('ip=${netX/ip} netmask=${netX/netmask} gateway=${netX/gateway} dns=${dns}')
+      if iface.tag.present? &&  iface.attached_to.present?
+        options.push("vlanid=#{iface.tag} ksdevice=#{iface.identifier}")
+      end
+    else
+        options.push("ip=${netX/ip}::${netX/gateway}:${netX/netmask}:#{hostname}:#{interface}:none nameserver=${dns}")
+        if iface.tag.present? &&  iface.attached_to.present?
+          options.push("vlan=#{iface.identifier}:#{iface.attached_to} bootdev=#{iface.identifier} ifname=#{iface.attached_to}:#{mac}")
+        end
+    end -%>
+
+<% ksoptions = options.join(' ')-%>
 
 <%- if @host.operatingsystem.name != 'Fedora' && @host.operatingsystem.major.to_i >= 7 && host_param_true?('fips_enabled') %>
 <%-   fips = 'fips=1' -%>
 <%- else -%>
 <%-   fips = '' -%>
 <%- end -%>
 
-kernel <%= "#{@host.url_for_boot(:kernel)}" %> initrd=initrd.img ks=<%= foreman_url('provision')%><%= static %> inst.stage2=<%= @host.operatingsystem.medium_uri(@host) %> <%= stage2 %> ksdevice=<%= @host.mac %> network kssendmac ks.sendmac inst.ks.sendmac <%= net_options %> <%= fips %>
+kernel <%= "#{@host.url_for_boot(:kernel)}" %> initrd=initrd.img ks=<%= foreman_url('provision')%><%= static %> inst.stage2=<%= @host.operatingsystem.medium_uri(@host) %> <%= stage2 %> ksdevice=<%= @host.mac %> network kssendmac ks.sendmac inst.ks.sendmac <%= ksoptions %> <%= fips %>
```

Comment:
You can probably avoid introducing `ksoptions` variable and just do `options.join(' ')` on the `kernel ...` line directly.

Reply:
I have used ksoptions so that the same option will be used in other templates as well e.g kickstart_default_pxelinux.erb file. Can i proceed with your suggestion or can we keep it as it is?

Reply:
It's not big deal, up to you.